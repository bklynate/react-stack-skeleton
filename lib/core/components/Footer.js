"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = Footer;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * The footer for the site.
 */
function Footer(_ref) {
  var otherProps = _objectWithoutProperties(_ref, []);

  return _react2.default.createElement(
    "footer",
    _extends({ className: "info" }, otherProps),
    _react2.default.createElement(
      "p",
      null,
      "Double-click to edit a todo"
    ),
    _react2.default.createElement(
      "p",
      null,
      "Created by ",
      _react2.default.createElement(
        "a",
        { href: "http://heyn.io" },
        "Nick Heyn"
      )
    ),
    _react2.default.createElement(
      "p",
      null,
      "Part of ",
      _react2.default.createElement(
        "a",
        { href: "http://todomvc.com" },
        "TodoMVC"
      )
    )
  );
}