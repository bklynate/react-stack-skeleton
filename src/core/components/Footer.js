/* @flow */
import React from 'react';

type Props = {

};

/**
 * The footer for the site.
 */
export default function Footer({ ...otherProps }: Props): React.Element<*> {
  return (
    <footer className="info" {...otherProps}>
      <p>Double-click to edit a todo</p>
      <p>Created by <a href="http://heyn.io">Nick Heyn</a></p>
      <p>Part of <a href="http://todomvc.com">TodoMVC</a></p>
    </footer>
  );
}
