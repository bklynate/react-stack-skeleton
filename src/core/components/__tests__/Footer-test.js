import React from 'react';
import renderer from 'react-test-renderer';

import Footer from '../Footer';

test('it displays the "NYI" message.', () => {
  const component = renderer.create(
    <Footer />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
