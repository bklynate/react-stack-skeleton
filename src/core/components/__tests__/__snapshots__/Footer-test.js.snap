exports[`test it displays the "NYI" message. 1`] = `
<footer
  className="info">
  <p>
    Double-click to edit a todo
  </p>
  <p>
    Created by 
    <a
      href="http://heyn.io">
      Nick Heyn
    </a>
  </p>
  <p>
    Part of 
    <a
      href="http://todomvc.com">
      TodoMVC
    </a>
  </p>
</footer>
`;
